_posts/2016-12-14-four-tips-sass-variables.markdown

```diff
@@ -0,0 +1,237 @@
+
+{% highlight sass %}
+$color-map: (
+    color1 : #F00,
+    color2 : #0F0,
+    color3 : #00F
+);
+{% endhighlight %}
+
+Similar to lists, `@each` loops are used to access the values. Developers can also access values by their specific keys. So for the different colors in this map, we could do this:
+
+{% highlight sass %}
+// We assign names to the properties and values in the map
+@each $name, $color in $color-map {
+    
+    // We can use them both for more detailed selectors in the loop
+    .#{$name}-background { background-color: $color; }
+}
+
+// We can also use $map-get to get a certain value
+.cool-element {
+    background-color: map-get($color-map, color1); // Returns #F00
+}
+{% endhighlight %}
+
+This syntax is similar to lists and only needs a few changes for the extra value. Since developers often want a specific string for only class names, Sass maps are often used more than lists.
+
+Those are the essentials to the different Sass variables in this post. You can get more complex by [nesting Sass lists and maps,](https://www.sitepoint.com/sass-maps-vs-nested-lists/), or [using conditionals within your loops](http://thesassway.com/intermediate/if-for-each-while), but that's a level of detail for another article. The rest of the tips only rely on these basics (mostly).
+
+### Know the Different Variables Scopes
+
+In another parallel to Javascript variables, Sass variables have different scopes. While there's only two official scopes, I like to think there's three for better structure. So these three scopes, going from largest to smallest, are as follows:
+
+#### Global Scope
+
+Global variables are the onces referenced the earliest in a collection of Sass files, since they're meant for use in all of them. File(s) with global variables will output no CSS, and just declare the global variables and what they're for. They contain universal and important values, like color, typography, responsive breakpoints, spacing units, and more.
+
+#### Single File Scope
+
+This is the scope that isn't technically isn't real, but I think is useful to pretend exists. Variables with a single file scope only exist in one sass file. An example: in the styles for a thumbnail component, there'd be variables for the thumbnails padding, border, layout widths, or others. The most important thing is these are declared at the file's start and used only in that sheet. This way it's easy to adjust important styles for it without running through dozens or hundreds of lines of code. Just go to that file's variables and change the values.
+
+Why isn't this a real scope? Because these variables could be used by every sass partial afterward. It's important not to do this, since it risks changes to one component accidentally affecting others and creating bugs. Single file variables can only be used in the files declaring them.
```

Comment:
"Single file variables can only be used..." -> "**should** only be used"? since it's a recommendation, with nothing enforcing it?